coder/coder
# feat: use sse for workspace page

resolves #4058 

Based on @f0ssel's [implementation of SSE](https://github.com/coder/coder/pull/4060) on the BE, we are now using SSE on the FE to get workspace and resource updates (instead of polling).

## Background
Previously, we fetched a workspace, template, and permissions, and then we started two separate polling states: we polled for workspace, and we polled for resources.

Since @f0ssel's implementation returns both workspace and resources in one object, we are able to simplify.

Now: we fetch a workspace, template, and permissions, and then we initialize an `EventSource` that returns workspace and workspace-related data as the server updates the client. 

![Screen Shot 2022-09-19 at 4 06 28 PM](https://user-images.githubusercontent.com/19142439/191107496-3c451879-e392-4174-af92-bc5ffc3cd4fc.png)

## To Test
Pull down the branch and open a workspace; make sure the workspace and resources load.
Open Chrome inspect and check out the network tab: you should be able to see a watch request that shows events streaming in:
![Screen Shot 2022-09-19 at 4 20 53 PM](https://user-images.githubusercontent.com/19142439/191109265-548a811e-bcd0-409f-9778-402ace0631fb.png)
